src/ibex_bluesky_core/devices/dae/dae_tcb_settings.py

```diff
@@ -0,0 +1,145 @@
+"""ophyd-async devices and utilities for the DAE time channel settings."""
+
+import xml.etree.ElementTree as ET
+from dataclasses import dataclass
+from enum import Enum
+from typing import Dict
+from xml.etree.ElementTree import tostring
+
+from bluesky.protocols import Locatable, Location
+from ophyd_async.core import AsyncStatus, Device, SignalRW
+
+from ibex_bluesky_core.devices import (
+    compress_and_hex,
+    dehex_and_decompress,
+    isis_epics_signal_rw,
+)
+from ibex_bluesky_core.devices.dae import (
+    convert_xml_to_names_and_values,
+    get_all_elements_in_xml_with_child_called_name,
+    set_value_in_dae_xml,
+)
+
+TIME_UNIT = "Time Unit"
+CALCULATION_METHOD = "Calculation Method"
+TIME_CHANNEL_FILE = "Time Channel File"
+
+
+class TimeUnit(Enum):
+    """Time unit for DAE TCB settings."""
+
+    MICROSECONDS = 0
+    NANOSECONDS = 1
+
+
+class CalculationMethod(Enum):
+    """Calculation method for DAE TCB settings."""
+
+    SPECIFY_PARAMETERS = 0
+    USE_TCB_FILE = 1
+
+
+class TimeRegimeMode(Enum):
+    """Time Regime Mode options for a single row."""
+
+    BLANK = 0
+    DT = 1
+    DTDIVT = 2
+    DTDIVT2 = 3
+    SHIFTED = 4
```

Comment:
These could probably benefit from a small comment, I have no idea what `DTDIVT2` means off the top of my head...